catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/transformer/CswRecordMap.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.spatial.ogc.csw.catalog.endpoint.transformer;
+
+import org.xml.sax.helpers.NamespaceSupport;
+
+/**
+ * Maps property names from a CSW query to catalog taxonomy
+ */
+public interface CswRecordMap {
+
+    /**
+     * Gets the catalog taxonomy attribute name for the property name,
+     * or returns the property name if not mapped. {@link NamespaceSupport}
+     * will be used to resolve namespaces. If {@link NamespaceSupport} is
+     * null, then {@link CswRecordMap#getProperty(String)} will be used.
```

Comment:
✏️ You should change the docs here to say that in the case of null the `getProperty` method that does not use `NamespaceSupport` _should_ be used. Conversely, you could make this a more complex interface with a default method and two methods to be implemented, but that's overkill for this case.